internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,336 @@
+}
+
+type secretManager struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+	Engine   string `json:"engine"`
+	Host     string `json:"host"`
+	Port     string `json:"port"`
+	DBName   string `json:"dbname"`
+}
+
+type ssmSecret struct {
+	Name  string
+	Value string
+}
+
+type localRunVars struct {
+	wkldName string
+	appName  string
+	envName  string
+}
+
+type localRunOpts struct {
+	localRunVars
+
+	deployedWkld       []string
+	wkldDeployedToEnvs map[string][]string
+	store              store
+	ws                 wsWlDirReader
+	prompt             prompter
+	ecsLocalClient     ecsLocalClient
+	deployStore        deployedEnvironmentLister
+	envVersionGetter   func(string) (versionGetter, error)
+}
+
+func newLocalRunOpts(vars localRunVars) (*localRunOpts, error) {
+	sessProvider := sessions.ImmutableProvider(sessions.UserAgentExtras("local run"))
+	defaultSess, err := sessProvider.Default()
+	if err != nil {
+		return nil, err
+	}
+
+	store := config.NewSSMStore(identity.New(defaultSess), ssm.New(defaultSess), aws.StringValue(defaultSess.Config.Region))
+	deployStore, err := deploy.NewStore(sessProvider, store)
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.Use(afero.NewOsFs())
+	if err != nil {
+		return nil, err
+	}
+	ecsLocalClient := ecs.New(defaultSess)
+	opts := &localRunOpts{
+		localRunVars: vars,
+
+		deployedWkld:       []string{},
+		wkldDeployedToEnvs: make(map[string][]string),
+		prompt:             prompt.New(),
+		store:              store,
+		ws:                 ws,
+		ecsLocalClient:     ecsLocalClient,
+		deployStore:        deployStore,
+	}
+	opts.envVersionGetter = func(envName string) (versionGetter, error) {
+		envDescriber, err := describe.NewEnvDescriber(describe.NewEnvDescriberConfig{
+			App:         opts.appName,
+			Env:         envName,
+			ConfigStore: opts.store,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("new environment compatibility checker: %v", err)
+		}
+		return envDescriber, nil
+	}
+	return opts, nil
+}
+
+func (o *localRunOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	if _, err := o.store.GetApplication(o.appName); err != nil {
+		return fmt.Errorf("get application %s: %w", o.appName, err)
+	}
+	return nil
+}
+
+func (o *localRunOpts) Ask() error {
+	if err := o.validateOrAskWorkloadName(); err != nil {
+		return err
+	}
+	if err := o.validateOrAskEnvName(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *localRunOpts) Execute() error {
+	taskDef, err := o.ecsLocalClient.TaskDefinition(o.appName, o.envName, o.wkldName)
+	if err != nil {
+		return fmt.Errorf("get task definition: %w", err)
+	}
+
+	awsSession := session.Must(session.NewSessionWithOptions(
+		session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		}))
+
+	_, err = o.getDecryptedSSMSecrets(taskDef, awsSession)
+	if err != nil {
+		fmt.Println("unable to get secrets from SSM store")
```

Comment:
We typically wrap an error only if it is returned from an external package:
```go
if err := otherPackage.AFunction(); err != nil { return fmt.Errorf("error from AFunction: %w", err) }
```

The wrapping gives a context of the error and gives the users/us an idea of where the error happens.

The "package-level" context is in most of the time enough to identify the place at which the error is returned. Wrap an intra-package error, in this case, usually just adds noise to the error message. Consider if we wrap every single error on every calls, then the error message would be super long and is basically a function call stack 😂 


```suggestion
		return err
```